src/copyright/agent/database.cc

```diff
@@ -300,21 +300,38 @@ bool CopyrightDatabaseHandler::insertNoResultInDatabase(long int agentId, long i
 
 bool CopyrightDatabaseHandler::insertInDatabase(DatabaseEntry& entry) const
 {
-  return dbManager.execPrepared(
-    fo_dbManager_PrepareStamement(
-      dbManager.getStruct_dbManager(),
-      "insertInDatabase",
-      "INSERT INTO "
-      IDENTITY
-      "(agent_fk, pfile_fk, content, hash, type, copy_startbyte, copy_endbyte)"
-        " VALUES($1,$2,$3,md5($3),$4,$5,$6)",
-      long, long, char*, char*, int, int
-    ),
-    entry.agent_fk, entry.pfile_fk,
-    entry.content.c_str(),
-    entry.type.c_str(),
-    entry.copy_startbyte, entry.copy_endbyte
-  );
+  if(!strcmp(entry.type.c_str(),IDENTITY) || strcmp(entry.type.c_str(),"statement")){
```

Comment:
### to explain my confusion:

You can rewrite this roughly in the following way (related to the [other comment below](https://github.com/fossology/fossology/pull/869/files#r126643170)):
```cc
if(  !strcmp(entry.type.c_str(),IDENTITY) // [1]
   || strcmp(entry.type.c_str(),"statement") // [2]
  ){
    tablename = "author";
} else { 
    tablename = IDENTITY;
}
// [...]
```

then
- the statement [1] says that "if you have to handle some entry whose type does not match your identity, then write the result to the `author`-table" and
- the statement [2] says something like "if you have to handle some entry of type `statement` then write it to the `author`-table"

In the case when `IDENTITY="copyright"` this means that you write everything to the `author` table, since either the type is never equal to `copyright` (i.e. is `statement` or something else for non-copyrights).

##

This result has not been verified in the database.